### PR TITLE
InformationFlow for Connectors

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [X] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #840 

### What is the new behavior?

- [X] Information flow is configurable from the property editor
- [x] Information flow is visible on the connector line
- [x] Information flow can be reversed

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by danyeaw** [comment]
@amolenaar I really like this implementation! Selecting from the PropertyPage was a really nice way to go about selecting the Information Items.

I did find what I think is a bug and I can reproduce it by:
1. Create two Properties
2. Create a connector between them
3. Enable the Information Flow using the Property Page
4. Don't select an Information Item
5. Click off the Connector and then reselect it

```
Traceback (most recent call last):
  File "C:\Users\dan\Projects\gaphor\gaphor\ui\elementeditor.py", line 155, in create_pages
    page = adapter.construct()
  File "C:\Users\dan\Projects\gaphor\gaphor\UML\components\componentspropertypage.py", line 94, in construct
    entry.set_text(self.subject.informationFlow[0].conveyed[0].name or "")
  File "C:\Users\dan\Projects\gaphor\gaphor\core\modeling\collection.py", line 76, in __getitem__
    return self.items.__getitem__(key)
  File "C:\Users\dan\Projects\gaphor\gaphor\core\modeling\listmixins.py", line 263, in __getitem__
    return super().__getitem__(key)  # type: ignore[misc] # noqa: F821
  File "C:\Users\dan\Projects\gaphor\gaphor\core\modeling\listmixins.py", line 93, in __getitem__
    return super().__getitem__(key)  # type: ignore[misc] # noqa: F821
IndexError: list index out of range
```

File: gaphor/UML/components/componentspropertypage.py [hunk 1]
```diff
@@ -41,3 +43,89 @@ def _on_ii_change(self, button, gparam):
...
+                    self._on_information_flow_name_changed,
+                ),
+                "invert-direction-changed": (self._invert_direction_changed,),
+            },
+        )
+
+        self.combo = combo = builder.get_object("information-flow-combo")
+        use_flow: Gtk.Switch = builder.get_object("use-information-flow")
+
+        for c in model.select(UML.Class):
```
Comment (danyeaw) [inline gaphor/UML/components/componentspropertypage.py:80]: Should we also add support for ValueTypes?

**Comment by amolenaar** [comment]
I'll fix the error. InformationFlow also needs an icon. 

We can add ValueTypes in a follow-up PR.

Should we also support ItemFlow? Then the flow can be based on a property. Only I'm not sure for which connections to use InformationFlow and for which ones to use ItemFlow.

**Comment by amolenaar** [comment]
@danyeaw Can this PR be merged?

**Review by danyeaw — APPROVED** [review]
Nice @amolenaar!

**Pull request merged**